src/contract-details/ContractDetailsCard.js

```diff
@@ -48,24 +49,25 @@ export default class ContractDetailsCard extends Component {
 						</tr>
 					</tbody>
 				</table>
-				{(validToSell && !validationError) ?
-					<button
-						className="sell-at-market-btn"
-						onClick={() => actions.sellContract(contract.contract_id, 0)}
-					>
-						Sell at Market (<NumberPlain
-							value={contract.bid_price}
-							currency={contract.currency}
-							isProfit={v => v - contract.buy_price}
-						/>)
-					</button> :
+				{(durationUnit !== 't') && (
```

Comment:
Have you check that this actually happens?

By right, if it's not allowed to be sold, is_valid_to_sell would be 0, thus validToSell would be false, 

It should not required an additional field check for this.